core/src/main/java/io/seata/core/rpc/ShutdownHook.java

```diff
@@ -91,35 +91,63 @@ public static void removeRuntimeShutdownHook() {
 
     private static class DisposablePriorityWrapper implements Comparable<DisposablePriorityWrapper>, Disposable {
 
+        private static AtomicLong seq = new AtomicLong();
+
         private Disposable disposable;
 
         private int priority;
 
+        private long seqId;
+
         public DisposablePriorityWrapper(Disposable disposable, int priority) {
             this.disposable = disposable;
             this.priority = priority;
+            this.seqId = seq.incrementAndGet();
         }
 
         @Override
         public int compareTo(DisposablePriorityWrapper disposablePriorityWrapper) {
-            return priority - disposablePriorityWrapper.priority;
```

Comment:
use PriorityQueue to replace TreeSet will solve this problem once and forever. No need to add a "seqId" to fix the probem which is actually caused by TreeSet.